components/ServiceProvider/index.js

```diff
@@ -22,114 +22,161 @@ const ServiceDetails = styled.div`
     setShowContactInfo(!showContactInfo);
   };
- 
+
   const handleEdit = (updatedServiceCard) => {
     setEditedCard(updatedServiceCard);
   };
-  
+
   const handleSave = (event) => {
-    event.preventDefault()
+    event.preventDefault();
 
     handleEditServiceCard(editedCard);
-    // Zurücksetzen der Service Card nach dem Speichern!
     setEditedCard(null);
   };
 
-return (
-  <ServiceProviderWrapper key={card.id}>
-        {editedCard && editedCard.id === card.id ? (
+  const handleDelete = () => {
+    const isConfirmed = window.confirm(
+      "Are you sure you want to delete this service provider?"
+    );
+
+    if (isConfirmed) {
+      const updatedCards = serviceCards.filter((cards) => cards.id !== card.id);
+      setServiceCards(updatedCards);
+    }
+  };
 
-            <form onSubmit={handleSave}>
-            <label htmlFor="firstName"> First Name: </label>
-            <input
-              type="text"
-              id="firstName"
-              name="firstName"
-              required
-              value={editedCard.firstName}
-              onChange={(event) => setEditedCard({ ...editedCard, firstName: event.target.value })}
-            />
-            <label htmlFor="lastName">Last Name: </label>
-            <input
-              type="text"
-              id="lastName"
-              name="lastName"
-              required
-              value={editedCard.lastName}
-              onChange={(event) => setEditedCard({ ...editedCard, lastName: event.target.value })}
-            />
-            <label htmlFor="skills">Skills: </label>
-            <input
-              type="text"
-              id="skills"
-              name="skills"
-              required
-              value={editedCard.skills}
-              onChange={(event) => setEditedCard({ ...editedCard, skills: event.target.value })}
-            />
-            <label htmlFor="needs">Needs: </label>
-            <input
-              type="text"
-              id="needs"
-              name="needs"
-              required
-              value={editedCard.needs}
-              onChange={(event) => setEditedCard({ ...editedCard, needs: event.target.value })}
-            />
-            <label htmlFor="email">email: </label>
-            <input
-              type="email"
-              id="email"
-              name="email"
-              required
-              value={editedCard.email}
-              onChange={(event) => setEditedCard({ ...editedCard, email: event.target.value })}
-              />
-            <label htmlFor="phone">phone: </label>
-            <input
-              type="tel"
-              id="phone"
-              name="phone"
-              required
-              value={editedCard.phone}
-              onChange={(event) => setEditedCard({ ...editedCard, phone: event.target.value })}
-            />
+  return (
+    <ServiceProviderWrapper key={card.id}>
+      {editedCard && editedCard.id === card.id ? (
```

Comment:
```suggestion
      { editedCard.id === card.id ? (
```

Shouldnt this be enough? editedCard.id cannot be equal to card.id is editedCard is false, isnt it?